src/modules/app/createAppContainer.js

```diff
@@ -100,14 +101,38 @@ const createCityContentNavigator = (params: CreateNavigationContainerParamsType)
   }
 }
 
-const createAppNavigationContainer = (params: CreateNavigationContainerParamsType): NavigationContainer<*, *, *> => {
+type NavigatorPropsType = {|
+  navigation: NavigationScreenProp<*>
+|}
+
+const createSwitchNavigatorWithSnackbar = (
+  params: CreateNavigationContainerParamsType
+): NavigationNavigator<*, *, *> => {
   const cityContentNavigator = createCityContentNavigator(params)
-  return createAppContainer(
-    createSwitchNavigator({
-      'Intro': IntroContainer,
-      'Landing': LandingContainer,
-      'CityContent': cityContentNavigator
-    }, { initialRouteName: params.initialRouteName }))
+  const SwitchNavigator = createSwitchNavigator({
+    'Intro': IntroContainer,
+    'Landing': LandingContainer,
+    'CityContent': cityContentNavigator
+  }, { initialRouteName: params.initialRouteName })
+
+  class SwitchNavigatorWithSnackbar extends React.Component<NavigatorPropsType> {
```

Comment:
I don't think it makes sense to move the class to a separate file without the method `createAppContainer` since the `SwitchNavigator` is used for the `static`s.